### DISCUSSION TITLE
COPC Validator with PDAL generated copc.laz file

### DISCUSSION THREAD

**Discussion by Hutch07** [copcio/copcio.github.io#66] [General]
I have generated several copc.laz files using a simple PDAL pipeline and the files seem to be OK when I open them with viewers like CloudCompare.   When I try and view it on https://validate.copc.io/ I am getting a **_Points out of bounds_** error.   When I look at the header and use Laspy to get the maxes and mins it looks like all points are within the header Max/Min.  This happens specifically with my files.  When I PDAL - copc another test file (Bunny) it seems to be OK.   What can I check on these files to troubleshoot what is happening?
Before and after files are here - https://drive.google.com/file/d/1gfZOO45pvfpJS2FYQITjRPfYDzAeDn7O/view?usp=sharing

**Comment by connormanning** [comment]
Thank you for providing both the input and the COPC data, we are investigating this and it is very helpful to have data that reproduces your issue.

I think I see what the problem is, but we need to have some discussion about it to determine whether it is a data issue or a validator issue, and what the fix would be in either case.

As far as I can tell, this issue can only occur if the input data is of higher precision than the output COPC data.  In your case, your input is scaled to 0.00025 but output is 0.01.  So I think if you set your `scale_x, scale_y, scale_z` options on your COPC writer ([doc](https://pdal.io/en/latest/stages/writers.copc.html)) to match your input then you will overcome this issue in the meantime.  You may want to set your `offset_x/y/z` values to `auto` as well.

**Comment by Hutch07** [comment]
I am trying this pipeline.
[
    "E:/SSI_Projects/2024_Projects/COPC_View/LAZ/OHM_Canton-Ohio_RGB (1004).laz",
    {
        "type":"writers.copc",
        "filename":"E:/SSI_Projects/2024_Projects/COPC_View/LAZ/Ohio_RGB2.copc.laz",
        "forward":"vlr",
        "scale_x":"0.00025", 
        "scale_y":"0.00025", 
        "scale_z":"0.00025", 
        "offset_x":"auto", 
        "offset_y":"auto", 
        "offset_z":"auto"
    }
]

**Comment by connormanning** [comment]
That pipeline looks good to me at a glance.  One note: this validation error isn't a big deal in practical terms.  It looks like you technically have 2 points that are slightly outside their advertised COPC node extents.  We are looking into resolving the underlying issue there, but this won't cause any real-world problems for you.

Your data looks fine to me - you can drag/drop it into https://viewer.copc.io and it works just fine.  Although for some reason it appears to be about 35m above the terrain models, you can adjust that on the `info` page of viewer.copc.io.

**Reply by Hutch07** [reply]
Thank you for your help.
